nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/GetMongo.java

```diff
@@ -149,10 +149,16 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
             }
         }
 
-        final Document query = getQuery(context, session, input );
+        final Document query;
+        try {
+            query = getQuery(context, session, input);
+        } catch (Exception ex) {
+            getLogger().error("Error parsing query.", ex);
+            if (input != null) {
+                session.transfer(input, REL_FAILURE);
+            }
 
-        if (query == null) {
-            return;
+            return; //We need to stop immediately.
```

Comment:
Just wondering if the `return` is really needed here?

Reply:
Yeah, we want onTrigger to stop immediately and commit things to REL_FAILURE.

Reply:
Fair point. I'm not having access to my machine right now. Will close this review by this week end.

Reply:
@joewitt any objections with me merging it since @zenfenan can't do it right now?